src/common/translations/localize.ts

```diff
@@ -54,84 +60,58 @@ export interface FormatsType {
   time: FormatType;
 }
 
-/**
- * Adapted from Polymer app-localize-behavior.
- *
- * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
- * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
- * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
- * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
- * Code distributed by Google as part of the polymer project is also
- * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
- */
-
-/**
- * Optional dictionary of user defined formats, as explained here:
- * http://formatjs.io/guides/message-syntax/#custom-formats
- *
- * For example, a valid dictionary of formats would be:
- * this.formats = {
- *    number: { USD: { style: 'currency', currency: 'USD' } }
- * }
- */
+const astCache: Record<string, ReturnType<typeof parse>> = {};
```

Comment:
this cache should be cleared everytime `computeLocalize` is called

Reply:
Actually no. The key is a translated value, and a string is guaranteed to always have the same AST when parsed.

Reply:
Correction, it should not be cleared everytime `computeLocalize` is called, but when the language is changed.

You don't want to keep stuff in cache that you don't even have resources for anymore, so will never be used 🤷‍♂️

Reply:
Ah, right, freeing up memory. I'll add that to my TODO list.